### PR TITLE
Add explanations to CSS colors

### PR DESCRIPTION
**Pull request by fahhem**
This is for the UI experiments framework, which parses CSS variables and comments to have explanations for colors to be modified.

This change is [Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)

### CODE REVIEW

**Comment by fahhem** [comment]
@earlAchromatic I think you made some of these tweaks, so if you can comment on the .. comments, that would be great!

**Review by pkaminski — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 4 unresolved discussions (waiting on @earlAchromatic and @fahhem)

___
*[`docs/tweaks.css` line 5 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_Dt_Ah2_pze5qeSO1X:-N_Dt_Ah2_pze5qeSO1Y:b6lskpl) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L5)):*
> ```CSS
>   content:attr(data-line-number);
>   position:absolute;
>   color: #444; /* line number background */
> ```

background -> color (foreground color) (I think, anyway)
___
*[`docs/tweaks.css` line 20 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_Du-cC5YpUYP8mu9N6:-N_Du-cC5YpUYP8mu9N7:b-xsbhna) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L20)):*
> ```CSS
> 
> .two.column.diff .diff.line:not(.keep) .line.wrapper:not(.empty) .left.line.content {
>   background-color: #FBDDDD;  /* removed line on the left */
> ```

These may all need to be updated to use the new `--diff-remove` and `--diff-add` variables instead.  @earlAchromatic would know best.
___
*[`docs/tweaks.css` line 24 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuFq5CTbzvr9FnF_2:-N_DuFq5CTbzvr9FnF_3:bd3nbwm) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L24)):*
> ```CSS
> .two.column.diff .diff.line.base:not(.keep) .line.wrapper:not(.empty) .left.line.content,
> .two.column.diff .diff.line.whitespace:not(.keep) .line.wrapper:not(.empty) .left.line.content {
>   background-color: #FFF7F7;  /* normal line on the left */
> ```

This is a line with only base or whitespace changes on the left.  It needs an even lighter background color because those deltas are themselves highlighted more lightly.  Same below.
___
*[`docs/tweaks.css` line 40 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuTG815LZLNrHBuaw:-N_DuTG815LZLNrHBuax:bakpacw) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L40)):*
> ```CSS
> .two.column.diff .diff.line.base .left.line.content .delta,
> .two.column.diff .diff.line.whitespace .left.line.content .delta {
>   background-color: rgba(240, 92, 92, 0.15); /* removed line whitespace */
> ```

or base-only

**Review by earlAchromatic — COMMENTED** [review]
Looks like Piotr has the comments on your comments covered. We need to get the snippets updated so they all play nice together though. I will update the spots in the app and the snippets with these changes.

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-:-N_FnLTRDPuzTgyuTW7C:bihk3li)* status: all files reviewed, 6 unresolved discussions (waiting on @fahhem and @pkaminski)

___
*[`docs/tweaks.css` line 19 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_FSpvACb9bsTx0zpY6:-N_FSpvB0DZTUOfmiPWB:b-7o9md8) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L19)):*
> ```CSS
> // ANCHOR: highlight-whole-line
> 
> .two.column.diff .diff.line:not(.keep) .line.wrapper:not(.empty) .left.line.content {
> ```

Not sure when this stopped working but this rule wasn't overriding the existing background rule so I had to change it to get it to work for edit lines. Same below.



_Suggestion:_
```CSS
wo.column.diff .diff.line:not(.keep) .line.wrapper.diffwidth:not(.empty) .left.line.content {
```
___
*[`docs/tweaks.css` line 20 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_Du-cC5YpUYP8mu9N6:-N_FX7LeB3VdpzkGcU5z:b6sv65d) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L20)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

These may all need to be updated to use the new `--diff-remove` and `--diff-add` variables instead.  @earlAchromatic would know best.
</blockquote></details>

This will work fine until you go to combine it with the blue/yellow diff color snippet.
So yes they will need updated but it's not as simple as just adding them in. We need to derive the snippet colors from the ones in the variables. In order to do this properly we have to split out the user defined variables a bit more and do some finessing so that the derived colors are just a lighter shade of whatever the user sets as their diff colors. 
These don't take into account the contrast variables either though that is a much simpler fix. 
___
*[`docs/tweaks.css` line 27 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_FTlCF2ZL0UqIHfbOW:-N_FTlCF2ZL0UqIHfbOX:b-ungxlm) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L27)):*
> ```CSS
> }
> 
> .two.column.diff .diff.line:not(.keep) .line.wrapper:not(.empty) .right.line.content {
> ```

Same as note above.



_Suggestion:_
```CSS
.two.column.diff .diff.line:not(.keep) .line.wrapper.diffwidth:not(.empty) .right.line.content {
```

**Review by fahhem — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @earlAchromatic and @pkaminski)

___
*[`docs/tweaks.css` line 19 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_FSpvACb9bsTx0zpY6:-N_Goz-U0wj2TSsOa_Jw:b-896fix) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L19)):*
<details><summary><i>Previously, earlAchromatic (Jacob Milhorn) wrote…</i></summary><blockquote>

Not sure when this stopped working but this rule wasn't overriding the existing background rule so I had to change it to get it to work for edit lines. Same below.
</blockquote></details>

Done.
___
*[`docs/tweaks.css` line 20 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_Du-cC5YpUYP8mu9N6:-N_GpWjJD05PoBY-cHMB:bfi0pww) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L20)):*
<details><summary><i>Previously, earlAchromatic (Jacob Milhorn) wrote…</i></summary><blockquote>

This will work fine until you go to combine it with the blue/yellow diff color snippet.
So yes they will need updated but it's not as simple as just adding them in. We need to derive the snippet colors from the ones in the variables. In order to do this properly we have to split out the user defined variables a bit more and do some finessing so that the derived colors are just a lighter shade of whatever the user sets as their diff colors. 
These don't take into account the contrast variables either though that is a much simpler fix. 
</blockquote></details>

what action can I take for this? Is this fine as-is but could be better, or just broken?
___
*[`docs/tweaks.css` line 27 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_FTlCF2ZL0UqIHfbOW:-N_GpNNo681A0p2kJlWv:b-896fix) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L27)):*
<details><summary><i>Previously, earlAchromatic (Jacob Milhorn) wrote…</i></summary><blockquote>

Same as note above.
</blockquote></details>

Done.

**Review by pkaminski — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 5 unresolved discussions (waiting on @earlAchromatic and @fahhem)

___
*[`docs/tweaks.css` line 24 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuFq5CTbzvr9FnF_2:-N_GuMM7-JJxv2P5GID-:b-uimi1o) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L24)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

This is a line with only base or whitespace changes on the left.  It needs an even lighter background color because those deltas are themselves highlighted more lightly.  Same below.
</blockquote></details>

"no changes is wrong -- it's "only base changes" ("no changes" wouldn't have any highlight at all)
___
*[`docs/tweaks.css` line 40 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuTG815LZLNrHBuaw:-N_GuB7X1NZ4cuJX77T_:bll6ask) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L40)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

or base-only
</blockquote></details>

not blank -- base-only

**Review by fahhem — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 5 unresolved discussions (waiting on @earlAchromatic and @pkaminski)

___
*[`docs/tweaks.css` line 24 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuFq5CTbzvr9FnF_2:-N_Gup8W-IdZP5_vywfW:b-swak6r) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L24)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

"no changes is wrong -- it's "only base changes" ("no changes" wouldn't have any highlight at all)
</blockquote></details>

sorry, but what does it mean to have base changes? Why does the base revision have changes? I feel like I'm missing something

**Comment by earlAchromatic** [comment]
*[`docs/tweaks.css` line 20 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_Du-cC5YpUYP8mu9N6:-N_GtWQU5zAyzhYZN9Df:b-a6daeo) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L20)):*
<details><summary><i>Previously, fahhem (Fahrzin Hemmati) wrote…</i></summary><blockquote>

what action can I take for this? Is this fine as-is but could be better, or just broken?
</blockquote></details>

After looking at your deploy, I realized it's fine since you give the user ability to set each of the colors individually in 'highlight-whole-line' option.  Since this is just a tool to help users get by in the meantime I'd say it's fine as is.

It's not great UX (people who don't want to mess with CSS probably also don't want to set 8 separate colors just for diffs) and if I rework those variables in the app and update the snippet the colors will be in sync and you can eliminate the highlight color options altogether.

**Review by pkaminski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 5 unresolved discussions (waiting on @earlAchromatic and @fahhem)

___
*[`docs/tweaks.css` line 24 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuFq5CTbzvr9FnF_2:-N_GwwT11tddzpFMSXbs:b-vakey2) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L24)):*
<details><summary><i>Previously, fahhem (Fahrzin Hemmati) wrote…</i></summary><blockquote>

sorry, but what does it mean to have base changes? Why does the base revision have changes? I feel like I'm missing something
</blockquote></details>

See the [docs](https://docs.reviewable.io/files.html#file-diff):

**Review by fahhem — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 4 unresolved discussions (waiting on @earlAchromatic and @pkaminski)

___
*[`docs/tweaks.css` line 24 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuFq5CTbzvr9FnF_2:-N_H2UIk-K8EroZyIJSg:b-896fix) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L24)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

See the [docs](https://docs.reviewable.io/files.html#file-diff):


</blockquote></details>

Done.
___
*[`docs/tweaks.css` line 40 at r1](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-N_DuTG815LZLNrHBuaw:-N_H2ttT8lwFV5bzYb7s:b-896fix) ([raw file](https://github.com/Reviewable/Reviewable/blob/ed487acc67a5e61496aadbaf3fd7aad9a3b92225/docs/tweaks.css#L40)):*
<details><summary><i>Previously, pkaminski (Piotr Kaminski) wrote…</i></summary><blockquote>

not blank -- base-only
</blockquote></details>

Done.

**Review by earlAchromatic — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087)* status: all files reviewed, 2 unresolved discussions (waiting on @pkaminski)

**Review by pkaminski — COMMENTED** [review]
:lgtm:

> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Reviewable/Reviewable/1087#-:-N_MQH6U702mxCNKEl0t:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @fahhem)

File: docs/tips.md [hunk 1]
```diff
@@ -60,15 +60,15 @@ A darker example:
 Don't like the merge bunny?  Turn it off like this:
 
 ```css
-{{#include tweaks.css:no-bunny}}
+{{#include tweaks.css:no-merge-bunny}}
```
Comment (fahhem) [inline docs/tips.md:63]: test

**Pull request merged**